src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerPathResolver.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+using System.Collections.Immutable;
+using System.Globalization;
+
+namespace Microsoft.CodeAnalysis.Remote.Diagnostics;
+
+/// <summary>
+/// For analyzers shipped in Roslyn, different set of assemblies might be used when running
+/// in-proc and OOP e.g. in-proc (VS) running on desktop clr and OOP running on ServiceHub .Net6
+/// host. We need to make sure to use the ones from the same location as the remote.
+/// </summary>
+internal sealed class RemoteAnalyzerPathResolver(string baseDirectory) : IAnalyzerPathResolver
```

Comment:
This was the `RemoteAnalyzerAssemblyLoader` type . Given that `AnalyzerAssemblyLoader` is `sealed` now it couldn't be a derivation anymore. The natural change was for it to be `IAnalyzerPathResolver`. At the same time I can find no where in our code base where this is used, except unit tests, so I'm not 100% sure what to do here. 

@jasonmalinowski

Reply:
Can dig further if we still need to.

Reply:
Agreed this appears to be dead code. Nothing is still consuming this in this new PR, right? Then I'd say just delete it -- if something is depending on this...we already broken them whether this stays in the new form or not.